### ISSUE TITLE
Failed to create sky apertures using source properties

### ISSUE THREAD

**Issue by shiaki** [astropy/photutils#692]
**Issue at:**
https://github.com/astropy/photutils/blob/6dd2739dc93caa058179ab4e9dacf82c9c1f8c21/photutils/utils/wcs_helpers.py#L74

**Environment:**
Ubuntu 16.04, 
Python (via Anaconda): `3.6.2 |Continuum Analytics, Inc.| (default, Jul 20 2017, 13:51:32) \n[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]`
astropy: v3.0.3
numpy: v1.14.5
photutils: v0.4

**Sample code to reproduce the problem:**
https://gist.github.com/shiaki/7b0b3333ba7203f47d3ed13f1bb32a76

**Error information:**
```
Traceback (most recent call last):
  File "fail.py", line 42, in <module>
    theta=props[0].orientation)
  File "/home/xxxxxx/anaconda3/envs/xxxxxx/lib/python3.6/site-packages/photutils/aperture/ellipse.py", line 371, in __init__
    assert_angle_or_pixel('a', a)
  File "/home/xxxxxx/anaconda3/envs/xxxxxx/lib/python3.6/site-packages/photutils/utils/wcs_helpers.py", line 78, in assert_angle_or_pixel
    "units".format(name))
ValueError: a should have angular or pixel units
```

**Description:**
The semi major/minor axes calculated by `source_properties` have units of `astropy.units.pixel` (L33 in the pasted code), but the values/units are not accepted when initializing sky apertures like `SkyEllipticalAperture`.

However, if I manually re-assign the units of semi major/minor axes to `pix` (L37, 38 in the pasted code), I can create sky apertures successfully.

The assertion `q.unit is u.pixel` at the highlighted line (L74) has failed, even the input values of `a` and `b` are in units of `pixel`, but cannot be recognized with `is` statement of python.

---
Thank for maintaining this very useful package! Please let me know if i can be of any help.

**Comment by shiaki** [comment]
I think a simple fix would be `if q.unit.physical_type == 'angle' or 'pix' in q.unit.name: `

This is probably a problem of astropy. We can reproduce the problem in a fairly easy way:
```
In [1]: import numpy as np
In [2]: import astropy.units as u

In [3]: dist_GC = 8.5 * u.kpc
In [4]: dist_GC.unit is u.kpc
Out[4]: True

In [5]: w = dist_GC ** 2
In [6]: w.unit
Out[6]: Unit("kpc2")

In [7]: x = np.sqrt(w)
In [8]: x.unit
Out[8]: Unit("kpc") # this is correct

In [9]: x.unit is u.kpc
Out[9]: False # but the assertion here fails.
```

When `SkyEllipticalAperture` calculates the semi-major/minor axes, it takes the square root of the first/second eigenvalues of the covariance matrix (which has units like `pix2`). The units are correctly propagated after `numpy.sqrt`, but the assertion of  `q.unit is u.pixel` would fail.

**Comment by larrybradley** [comment]
@shiaki On second thought, I don't think this is an upstream bug.  Instead of using `is`, the correct test should simply be the `==` operator, i.e. `x.unit == u.kpc` in your above example.  I'll fix the unit test in `photutils`.

**Comment by shiaki** [comment]
Hi @larrybradley , thanks for your reply and rapid fix! I have further investigated into the problem. This is probably an expected behavior of python (object identity vs. equality). After taking the square root, the derived unit `pix` (from `pix2`) has a different identity with object `u.pixel` in the module. I did not see official recommendation in astropy docs, but I have tested that `==` operator works as expected.
